app/services/copy_profiles_to_policies.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# A service class to copy Profile data to new Policy objects
+class CopyProfilesToPolicies
+  class << self
+    def run!
+      create_policies
+    end
+
+    private
+
+    def profiles
+      Profile.canonical(false)
+             .where(policy: nil)
+    end
+
+    def create_policies
+      profiles.where(external: false).find_each do |profile|
```

Comment:
@akofink The creation of policies and assignment of profiles seems incorrect. 

First problem:
With this migration it would be creating a single Policy for each Profile (from the set; non-canonical without a policy). First it gets all profiles that need don't have policy, for each it will create a policy and then it would reassign all other account's profiles that do not have a policy to newly created Policy. Since this processes it in batches, it would create redundant Policies. Some might become orphans (might be deleted by the hook).
This needs a bit more thinking (if it is a real problem). It might group Profiles by account, ref_id and os_major_version.

Second:
Grouping of profiles to a policy happens non-deterministically. I'd suggest to order the profiles by `created_at DESC` to create policies for latest profiles first. Otherwise policies would inherit attributes from a random profile.

Third:
The grouping to a profile happens regardless of an OS major version. The `ref_id` of a policy is found across OS major versions. The Policy should be only for a single OS major version.

Reply:
Being handled in PR #610

Reply:
Whoops, sorry for including this :see_no_evil: